### PR TITLE
PCHR-841: Job contract reason for end

### PR DESCRIPTION
**Pull request by coldrunKacper**
- Adding new 'end_reason' field for HRJobContract Details entity.
- Creating new 'hrjc_contract_end_reason' Option Group containing three Contract End reasons.
- Adding the 'hrjc_contract_end_reason' Option Group to the CiviCRM Administration menu (under Dropdown Options).

Job Contract Summary page (before):
job-contract-summary-before
Job Contract Summary page (after) - notice there is a new 'End reason' label under 'Contract Dates' section:
job-contract-summary-after
Job Contract Details page (before):
job-contract-details-before
Job Contract Details page (after) - notice there is a new 'Contract End Reason' dropdown field after setting Contract End Date:
job-contract-details-after

### CODE REVIEW

File: hrjobcontract/js/src/job-contract/controllers/form/form-general.js [hunk 1]
```diff
@@ -52,8 +52,14 @@ define([
             });
 
             $scope.$watch('entity.details.period_end_date', function(){
-                $scope.dpDateStartMax = entityDetails.period_end_date ? moment(entityDetails.period_end_date).subtract(1, 'day').format() : '';
+                if (entityDetails.period_end_date) {
+                    $scope.dpDateStartMax = moment(entityDetails.period_end_date).subtract(1, 'day').format();  
+                } else {
+                    $scope.dpDateStartMax = '';
```
Comment (AkA84) [inline hrjobcontract/js/src/job-contract/controllers/form/form-general.js]: Couldn't this be set to `null`? Same for line below

File: hrjobcontract/js/src/job-contract/controllers/form/form-general.js [hunk 2]
```diff
@@ -52,8 +52,14 @@ define([
...
+                    entityDetails.end_reason = '';
+                }
                 $scope.duration = duration(entityDetails.period_start_date, entityDetails.period_end_date);
+
```
Comment (AkA84) [inline hrjobcontract/js/src/job-contract/controllers/form/form-general.js]: Unnecessary new line. I would actually put it between the if and this assignment
```js
}

$scope.duration = // ...

**Comment by AkA84** [comment]
PR description is missing, please add screenshots as you are creating a new page and you are amending an already existing template

File: hrjobcontract/CRM/Hrjobcontract/Upgrader.php [hunk 3]
```diff
@@ -1108,6 +1108,64 @@ function upgrade_1010() {
...
   }
+  
+  /**
+   * Adding 'end_reason' field for Job Contract Details,
+   * 'hrjc_contract_end_reason' Option Group with three Option Values 
+   * and Administration Menu entry for managing the Option Group values.
+   * 
+   * @return boolean TRUE
+   */
+  function upgrade_1011() {
```
Comment (nishant-bhorodia) [inline hrjobcontract/CRM/Hrjobcontract/Upgrader.php]: @coldrunKacper Please can we add a few comments to separate different sections in this function ? Thanks !!

**Comment by AkA84** [comment]
Approved (frontend)

**Comment by AkA84** [comment]
Approved on behalf of @nishant-bhorodia

**Pull request merged**